lib/sles4sap.pm

```diff
@@ -135,6 +136,49 @@ undef by default. Test modules testing for SAP ASE should set this property befo
 
 has ASE_RESPONSE_FILE => undef;
 
+=head2 download_hana_assets_from_server
+
+    $self->download_hana_assets_from_server()
+
+Download and extract HANA installation media to /sapinst directory of the SUT.
+The media location must be provided as ASSET_0 in the job settings and be
+available as an uncompressed tar in the factory/other directory of the openQA
+server
+
+=cut
+
+sub download_hana_assets_from_server {
+    my %params = @_;
+    my $target = $params{target} // '/sapinst';
+    my $nettout = $params{nettout} // 2700;
+    script_run "mkdir $target";
+    assert_script_run "cd $target";
+    my $filename = get_required_var('ASSET_0');
+    my $hana_location = data_url('ASSET_0');
+    # Each HANA asset is about 16GB. A ten minute timeout assumes a generous
+    # 27.3MB/s download speed. Adjust according to expected server conditions.
+    # checks if asset is already downloaded
+    my $asset_lock = "/tmp/asset_0";
+    my $asset_lock_found = script_run "test -e $asset_lock";    # 0 if asset is already downloaded
+    if ($asset_lock_found) {
+        assert_script_run "wget -O - $hana_location | tar -xf -", timeout => $nettout;
+        script_run "touch $asset_lock";
```

Comment:
Better run this one with `assert_script_run`

Reply:
done